packages/server/src/api/controllers/row/utils/basic.ts

```diff
@@ -111,8 +111,15 @@ export function fixArrayTypes(row: Row, table: Table) {
       try {
         row[fieldName] = JSON.parse(row[fieldName])
       } catch (err) {
-        // couldn't convert back to array, ignore
-        delete row[fieldName]
+        // Handling deprecated  single user type
+        const isDeprecatedSingleUser =
+          schema.type === FieldType.BB_REFERENCE &&
+          schema.subtype === BBReferenceFieldSubType.USER &&
+          schema.constraints?.type !== "array"
```

Comment:
I've seen this check in a couple of places, do we want to extract it out into `shared-core`?

Reply:
I actually thought about this. Refactoring it!